### PR TITLE
Add Reactor test helper functions #85

### LINKED ISSUE #85: Add Reactor test helper functions
Add `assertTransition()` and `assertFinish()` helper functions.

### PR DESCRIPTION
**Pull request by qrazhan**
Closes #85

### CODE REVIEW

File: workflow-rx2/src/main/java/com/squareup/workflow/rx2/TestUtils.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Square Inc.
```
Comment (zach-klippenstein) [inline workflow-rx2/src/main/java/com/squareup/workflow/rx2/TestUtils.kt]: Should be 2018.

File: workflow-rx2/src/main/java/com/squareup/workflow/rx2/TestUtils.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (zach-klippenstein) [inline workflow-rx2/src/main/java/com/squareup/workflow/rx2/TestUtils.kt]: Let's make a new module called `workflow-test` and put this there, so it doesn't pollute non-test code. I think we will probably end up putting more stuff in there at some point.

@rjrjr WDYT?

**Comment by rjrjr** [comment]
Definitely

File: sample-game-common/build.gradle [hunk 3]
```diff
@@ -38,4 +38,5 @@ dependencies {
   testImplementation deps.test.hamcrestCore
   testImplementation deps.test.junit
   testImplementation deps.rxjava2.extensions
+  testImplementation project(':workflow-test')
```
Comment (zach-klippenstein) [inline sample-game-common/build.gradle]: Project dependencies should go above any third-party ones in each group.

**Comment by qrazhan** [comment]
Gunna just merge as-is, since I don't have a ton of experience with co-routines. It'd be great to pair on the conversion so I can learn more! : )

File: workflow-rx2/src/main/java/com/squareup/workflow/rx2/TestUtils.kt (hunk 2 above)
Comment (rjrjr) [reply workflow-rx2/src/main/java/com/squareup/workflow/rx2/TestUtils.kt]: Yes yes yes.

**Pull request merged**